# CISC7204 Assignment 01 Module 01 Lab 02

**Student:** LU MENGJIE  
**Student ID:** MC643920  
**Topic:** Importing Data Sets Laptop Pricing

This notebook was completed in a local Python 3 Jupyter environment. The original JupyterLite download helper is skipped because `laptop_pricing_dataset_base.csv` is supplied in the same folder. The code and outputs below document Tasks 1–6.


## Evidence map

- Task 1: load the raw dataset and print the first five entries
- Task 2: assign the specified headers and print the first ten rows
- Task 3: replace `?` with `NaN` and report the missing-value counts
- Task 4: print the dataframe data types
- Task 5: print the complete statistical description
- Task 6: print the dataframe summary information

Each task is followed by a short explanation of what its output proves.


<p style="text-align:center">
    <a href="https://skills.network" target="_blank">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png" width="300" alt="Skills Network Logo">
    </a>
</p>


# **Hands-on Practice Lab: Importing Dataset - Laptops Pricing**

Estimated time needed: **20** minutes

In this lab, you will practice the process of loading and drawing basic insights on a dataset as learnt through the module. You are being provided with a fresh dataset on 'Laptop Pricing' which will be used for all the practice labs throughout the course.


# Objectives

After completing this lab you will be able to:

 - Import a dataset from a CSV file to a Pandas dataframe
 - Develop some basic insights about the dataset


# Setup


For this lab, we will be using the following libraries:

* `skillsnetwork` for downloading the daataset

*   [`pandas`](https://pandas.pydata.org/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMML0187ENSkillsNetwork31430127-2021-01-01) for managing the data.
*   [`numpy`](https://numpy.org/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMML0187ENSkillsNetwork31430127-2021-01-01) for mathematical operations.


### Importing Required Libraries


In [1]:
import pandas as pd
import numpy as np
if hasattr(pd.options, 'future') and hasattr(pd.options.future, 'infer_string'):
    pd.options.future.infer_string = False


The functions below will download the dataset into your browser:


In [2]:
from pathlib import Path
assert Path('laptop_pricing_dataset_base.csv').exists(), 'laptop_pricing_dataset_base.csv must be in the notebook folder'
print('Local dataset found: laptop_pricing_dataset_base.csv')


Local dataset found: laptop_pricing_dataset_base.csv


In [3]:
file_path = 'laptop_pricing_dataset_base.csv'


To obtain the dataset, utilize the download() function as defined above:


In [4]:
file_name = 'laptop_pricing_dataset_base.csv'
print(f'Using local dataset: {file_name}')


Using local dataset: laptop_pricing_dataset_base.csv


In [5]:
df = pd.read_csv(file_name, header=None)


> Note: This version of the lab is working on JupyterLite, which requires the dataset to be downloaded to the interface. While working on the downloaded version of this notebook on their local machines, the learners can simply **skip the steps above**, and simply use the URL directly in the `pandas.read_csv()` function. You can uncomment and run the statements in the cell below.


In [6]:
df = pd.read_csv(file_name, header=None)
print('Raw dataframe shape:', df.shape)


Raw dataframe shape: (238, 12)


<h1> Task #1: </h1>
<h3>Load the dataset to a pandas dataframe named 'df'</h3>
Print the first 5 entries of the dataset to confirm loading.


In [7]:
# Task 1: load the dataset and print the first five entries.
print(df.head())


     0   1          2   3   4   5       6    7   8    9     10    11
0  Acer   4  IPS Panel   2   1   5   35.56  1.6   8  256   1.6   978
1  Dell   3    Full HD   1   1   3  39.624  2.0   4  256   2.2   634
2  Dell   3    Full HD   1   1   7  39.624  2.7   8  256   2.2   946
3  Dell   4  IPS Panel   2   1   5  33.782  1.6   8  128  1.22  1244
4    HP   4    Full HD   2   1   7  39.624  1.8   8  256  1.91   837


**Explanation:** The displayed first five rows confirm that the raw laptop pricing file was loaded into the pandas dataframe named `df`, answering Task 1.


<details><summary>Click here for solution</summary>

```python
df = pd.read_csv(file_name, header=None)
print(df.head())
```
</details>


<h1> Task #2: </h1>
<h3>Add headers to the dataframe</h3>
The headers for the dataset, in sequence, are "Manufacturer", "Category", "Screen", "GPU", "OS", "CPU_core",
"Screen_Size_inch", "CPU_frequency", "RAM_GB", "Storage_GB_SSD", "Weight_kg" and "Price".

Confirm insertion by printing the first 10 rows of the dataset.


In [8]:
headers = ['Manufacturer', 'Category', 'Screen', 'GPU', 'OS', 'CPU_core', 'Screen_Size_inch', 'CPU_frequency', 'RAM_GB', 'Storage_GB_SSD', 'Weight_kg', 'Price']
df.columns = headers
print(df.head(10))


  Manufacturer  Category     Screen  GPU  OS  CPU_core Screen_Size_inch  \
0         Acer         4  IPS Panel    2   1         5            35.56   
1         Dell         3    Full HD    1   1         3           39.624   
2         Dell         3    Full HD    1   1         7           39.624   
3         Dell         4  IPS Panel    2   1         5           33.782   
4           HP         4    Full HD    2   1         7           39.624   
5         Dell         3    Full HD    1   1         5           39.624   
6           HP         3    Full HD    3   1         5           39.624   
7         Acer         3  IPS Panel    2   1         5             38.1   
8         Dell         3    Full HD    1   1         5           39.624   
9         Acer         3  IPS Panel    3   1         7             38.1   

   CPU_frequency  RAM_GB  Storage_GB_SSD Weight_kg  Price  
0            1.6       8             256       1.6    978  
1            2.0       4             256       2.2    

**Explanation:** The first ten rows show that the twelve specified headers were assigned in the required order, answering Task 2.


<details><summary>Click here for solution</summary>

```python
# create headers list
headers = ["Manufacturer", "Category", "Screen", "GPU", "OS", "CPU_core", "Screen_Size_inch", "CPU_frequency", "RAM_GB", "Storage_GB_SSD", "Weight_kg", "Price"]
df.columns = headers
print(df.head(10))
```
</details>


<h1> Task #3: </h1>
<h3>Replace '?' with 'NaN'</h3>
Replace the '?' entries in the dataset with NaN value, recevied from the Numpy package.


In [9]:
df.replace('?', np.nan, inplace=True)
print('Missing values by column after replacement:')
print(df.isna().sum())


Missing values by column after replacement:
Manufacturer        0
Category            0
Screen              0
GPU                 0
OS                  0
CPU_core            0
Screen_Size_inch    4
CPU_frequency       0
RAM_GB              0
Storage_GB_SSD      0
Weight_kg           5
Price               0
dtype: int64


**Explanation:** The replacement step converts the dataset's `?` markers to `NaN`; the printed counts provide visible evidence of the resulting missing values, answering Task 3.


<details><summary>Click here for solution</summary>

```python
df.replace('?',np.nan, inplace = True)
```
</details>


<h1> Task #4: </h1>
<h3>Print the data types of the dataframe columns</h3>
Make a note of the data types of the different columns of the dataset.


In [10]:
print(df.dtypes)


Manufacturer         object
Category              int64
Screen               object
GPU                   int64
OS                    int64
CPU_core              int64
Screen_Size_inch     object
CPU_frequency       float64
RAM_GB                int64
Storage_GB_SSD        int64
Weight_kg            object
Price                 int64
dtype: object


**Explanation:** The printed Series identifies the data type of every laptop dataframe column, answering Task 4.


<details><summary>Click here for solution</summary>

```python
print(df.dtypes)
```
</details>


<h1> Task #5: </h1>
<h3>Print the statistical description of the dataset, including that of 'object' data types.</h3>


In [11]:
print(df.describe(include='all'))


       Manufacturer    Category   Screen         GPU          OS    CPU_core  \
count           238  238.000000      238  238.000000  238.000000  238.000000   
unique           11         NaN        2         NaN         NaN         NaN   
top            Dell         NaN  Full HD         NaN         NaN         NaN   
freq             71         NaN      161         NaN         NaN         NaN   
mean            NaN    3.205882      NaN    2.151261    1.058824    5.630252   
std             NaN    0.776533      NaN    0.638282    0.235790    1.241787   
min             NaN    1.000000      NaN    1.000000    1.000000    3.000000   
25%             NaN    3.000000      NaN    2.000000    1.000000    5.000000   
50%             NaN    3.000000      NaN    2.000000    1.000000    5.000000   
75%             NaN    4.000000      NaN    3.000000    1.000000    7.000000   
max             NaN    5.000000      NaN    3.000000    2.000000    7.000000   

       Screen_Size_inch  CPU_frequency 

**Explanation:** The complete `describe(include='all')` table summarizes both numeric and categorical columns, including counts, unique values, most frequent values, and numerical statistics, answering Task 5.


<details><summary>Click here for solution</summary>

```python
print(df.describe(include='all'))
```
</details>


<h1> Task #6: </h1>
<h3>Print the summary information of the dataset.</h3>


In [12]:
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 238 entries, 0 to 237
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Manufacturer      238 non-null    object 
 1   Category          238 non-null    int64  
 2   Screen            238 non-null    object 
 3   GPU               238 non-null    int64  
 4   OS                238 non-null    int64  
 5   CPU_core          238 non-null    int64  
 6   Screen_Size_inch  234 non-null    object 
 7   CPU_frequency     238 non-null    float64
 8   RAM_GB            238 non-null    int64  
 9   Storage_GB_SSD    238 non-null    int64  
 10  Weight_kg         233 non-null    object 
 11  Price             238 non-null    int64  
dtypes: float64(1), int64(7), object(4)
memory usage: 22.4+ KB


**Explanation:** The `info()` output confirms the dataframe's 238 entries, 12 named columns, non-null counts, data types, and memory usage, answering Task 6.


<details><summary>Click here for solution</summary>

```python
print(df.info())
```
</details>


---


# Congratulations! You have completed the lab


## Authors


[Abhishek Gagneja](https://www.coursera.org/instructor/~129186572)

[Vicky Kuo](https://author.skills.network/instructors/vicky_kuo)


Copyright © 2023 IBM Corporation. All rights reserved.


<!--## Change Log


<!--|Date (YYYY-MM-DD)|Version|Changed By|Change Description|
|-|-|-|-|
|2023-09-15|0.1|Abhishek Gagneja|Initial Version Created|
|2023-09-18|0.2|Vicky Kuo|Reviewed and Revised|--!>
